tests/models/test_genericconfig.py

```diff
@@ -35,36 +64,100 @@ class SomeSubclass(GenericConfig):
     assert instance.description == instance.__doc__
 
 
-def test_attribute_get_item(generic_config: "GenericConfig") -> None:
+def test_attribute_get_item(generic_config: "CustomConfig") -> None:
     """Test configuration.__getitem__."""
     assert generic_config.configuration["integer"] == 5
 
 
-def test_attribute_set(generic_config: "GenericConfig") -> None:
-    """Test configuration.__setitem__, and thus also
-    configuration.__setattr__.
+def test_attribute_get_item_fail(generic_config: "CustomConfig") -> None:
+    """Ensure KeyError is raised for a non-existent key."""
+    non_existent_key = "non_existent_key"
+    assert non_existent_key not in generic_config.configuration
+    with pytest.raises(KeyError):
+        generic_config.configuration[  # pylint: disable=pointless-statement
+            non_existent_key
+        ]
+
+
+def test_attribute_set(generic_config: "CustomConfig") -> None:
+    """Test configuration.__setitem__, and thus also configuration.__setattr__.
+
+    Assign different values to test dynamic type-casting.
+    If a field has been specified to have a specific type, the incoming value should be
+    cast to this type. However, if it is a non-specified type (similar to a standard
+    dict key/value-pair), then it should be fine to change the value type for that
+    given key.
 
-    Assign both a valid and an invalid value, to test dynamic
-    type-checking.
     """
     generic_config.configuration["string"] = "bar"
-    try:
-        generic_config.configuration["string"] = 3.14
-        assert False
-    except TypeError:
-        pass
+    generic_config.configuration["string"] = 3.14
+    assert generic_config.configuration["string"] == str(3.14)
 
+    generic_config.configuration["float"] = "0"
+    assert generic_config.configuration["float"] == "0"
```

Comment:
The idea here is to check that if a field/key has not been defined using `pydantic.Field` it will _not_ be cast to any given type. This is equivalent to normal `dict` behaviour, which is what we want :)